substrate/client/tracing/src/logging/fast_local_time.rs

```diff
@@ -88,24 +100,59 @@ impl FormatTime for FastLocalTime {
 			let mut cache = cache.borrow_mut();
 
 			// Regenerate the timestamp only at most once each second.
-			if cache.last_regenerated_at != unix_time {
-				let ts = chrono::Local::now();
-				let fractional = (ts.nanosecond() % 1_000_000_000) / 1_000_000;
-				cache.last_regenerated_at = unix_time;
-				cache.last_fractional = fractional;
-				cache.buffer.length = 0;
-
-				write!(
-					&mut cache.buffer,
-					"{:04}-{:02}-{:02} {:02}:{:02}:{:02}.{:03}",
-					ts.year(),
-					ts.month(),
-					ts.day(),
-					ts.hour(),
-					ts.minute(),
-					ts.second(),
-					fractional
-				)?;
+			if cache.last_regenerated_at != unix_time || cache.last_utc != self.utc {
+				cache.last_utc = self.utc;
+
+				if self.utc {
+					let ts = chrono::Utc::now();
+					let fractional = (ts.nanosecond() % 1_000_000_000) / 1_000;
+					cache.last_regenerated_at = unix_time;
+					cache.last_fractional = fractional;
+					cache.buffer.length = 0;
+
+					write!(
+						&mut cache.buffer,
+						"{:04}-{:02}-{:02}T{:02}:{:02}:{:02}.{:06}Z",
+						ts.year(),
+						ts.month(),
+						ts.day(),
+						ts.hour(),
+						ts.minute(),
+						ts.second(),
+						fractional
+					)?;
+				} else {
+					let ts = chrono::Local::now();
+					let fractional = (ts.nanosecond() % 1_000_000_000) / 1_000_000;
+					cache.last_regenerated_at = unix_time;
+					cache.last_fractional = fractional;
+					cache.buffer.length = 0;
+
+					write!(
+						&mut cache.buffer,
+						"{:04}-{:02}-{:02} {:02}:{:02}:{:02}.{:03}",
+						ts.year(),
+						ts.month(),
+						ts.day(),
+						ts.hour(),
+						ts.minute(),
+						ts.second(),
+						fractional
+					)?;
+
+					if cache.timezone_offset.length == 0 {
+						write!(&mut cache.timezone_offset, "{}", ts.offset())?;
+					}
```

Comment:
In theory the current timezone can change, so this should not assume that the timezone stays constant.

This is only done once per second, so its fine to do this unconditionally.